### PR TITLE
Hammelmann/add density target

### PR DESCRIPTION
**Pull request by JanHammelmann**
Add density test in the box and in the collider mode as a new target.

### CODE REVIEW

File: test/densities/collider_setup/config.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# SMASH-1.5
+# Branch   : HEAD
+# System   : Linux-4.4.0-151-generic
+# Compiler : GNU 5.4.0
+# Build    : RelWithDebInfo
+# Date     : 2020-04-16T11:34:20
```
Comment (akschaefer) [inline test/densities/collider_setup/config.yaml:6]: Looks like you copied the output config, so better remove these lines and adjust the formatting of the rest to have the 4 character indentation.

File: test/densities/box_setup/particles_pi_k_n_lambda.txt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+########## Δ baryons ###################
+
+Δ       1.232 0.117 +   1114    2114    2214    2224
+
+########## Λ baryons ###################
+
+Λ        1.116 0      +     3122
+
+########## Σ baryons ###################
+Σ       1.189 0     +    3112    3212    3222
```
Comment (akschaefer) [inline test/densities/box_setup/particles_pi_k_n_lambda.txt:32]: Why are there Delta and Sigma baryons in the `pi_K_N_lambda` box? Also why is there a Delta decay in the Decaymodes list? Or should the target rather be a `pi_K_N_lambda_Delta_Sigma` target?

File: test/densities/plot_density_central_cell.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    import comp_to_prev_version as cpv
+    processes = []
+    cpv.plot_previous_results('densities', setup, '/' + DataOutfile.split('/')[-1])
+
+# old version ?
+plt.plot(TimeSteps, dens_arr, linewidth=2, linestyle="-", label='AuAu')
+plt.text(0.3, 4.5, " SMASH code:      %s\n SMASH analysis: %s\n" % \
+            (smash_version, smash_analysis_version), \
+            color = "gray", fontsize = 7)
+plt.title(r'AuAu, central cell, $E_{} = {}$ AGeV'.format('{Kin}', Energy))
```
Comment (akschaefer) [inline test/densities/plot_density_central_cell.py:79]: Nit-picky detail ... The AuAu is now present twice: In the title and in the legend. Could you modify the legend label to 'AuAu @ Ekin = 1.23' and set the title to only 'central cell'? Then you can also print the SMASH and SMASHana version in smaller fonts to the upper right to have it uniform across all targets again. Just check how it's done for the other plotting scripts...

**Review by akschaefer — CHANGES_REQUESTED** [review]
Very nice work :)! The target runs smoothly through and produces the desired output. However there are a couple of inconsistencies with respect to the other smash analysis targets which should be fixed and modified. Also, I believe the CMakeLists can be simplified. I aded the the corresponding comments to the PR. Let me know what you think or if we need to discuss it further.

File: test/densities/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,217 @@
+project(densities NONE)
+
+set(dens_box_sims "")
+set(dens_box_plots "")
+set(dens_collider_sims "")
+set(dens_collider_plots "")
+
+function(run_box_test
+         test_name
+         yaml_op_raw
```
[the remark this answers is not in the document]
Comment (JanHammelmann) [reply test/densities/CMakeLists.txt:10]: I thought about that as well. However when I try to remove this parameter, cmake crashes and I have no idea why.

File: test/densities/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+         var_list
```
[the remark this answers is not in the document]
Comment (JanHammelmann) [reply test/densities/CMakeLists.txt:11]: Good point! I changed it such that one iterates via a loop and you only give the total number of folders.

File: test/densities/box_setup/particles_pi_k_n_lambda.txt (hunk 2 above)
Comment (JanHammelmann) [reply test/densities/box_setup/particles_pi_k_n_lambda.txt:32]: Somehow in combination with the decaymodes these particles are needed so I added them... For this purpose it shouldn't make a difference. Maybe ```pi_K_N+Res``` is more suitable? ```pi_K_N_lambda_Delta_Sigma``` is a bit unhandy ;)

File: test/densities/CMakeLists.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (akschaefer) [reply test/densities/CMakeLists.txt:11]: Nice 👍

File: test/densities/CMakeLists.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (akschaefer) [reply test/densities/CMakeLists.txt:10]: Hm, surprising. I just tried it and it works fine for me. If you want, I can push a fix to your branch that removes this parameter...

**Review by akschaefer — APPROVED** [review]
Looks good now 👍

File: test/densities/box_setup/particles_pi_k_n_lambda.txt (hunk 2 above)
Comment (akschaefer) [reply test/densities/box_setup/particles_pi_k_n_lambda.txt:32]: I added a corresponding comment in the particle lists.txt. Those resonances are needed for decays of the others, that's why they are in there.

**Comment by akschaefer** [comment]
As discussed I made some more changes on your branch, regarding the removal of the `yaml_op_raw` parameter and some further cleanups in the CMakeLists, a bugfix for the plotting with the comparison to the previous version and some small cleanups for the plots themselves. Have a look at them and feel free to merge, if you agree.

**Comment by JanHammelmann** [comment]
Thanks for your changes!

**Pull request merged**